src/WebJobs.Script/Workers/WorkerConcurrencyManager.cs

```diff
@@ -27,40 +27,38 @@ internal class WorkerConcurrencyManager : IHostedService, IDisposable
         private readonly ILogger _logger;
         private readonly IFunctionInvocationDispatcherFactory _functionInvocationDispatcherFactory;
         private readonly IEnvironment _environment;
-        private readonly IFunctionsHostingConfiguration _functionsHostingConfigurations;
         private readonly IApplicationLifetime _applicationLifetime;
         private readonly long _memoryLimit = AppServicesHostingUtility.GetMemoryLimitBytes();
+        private readonly IOptions<FunctionsHostingConfigOptions> _functionsHostingConfigOptions;
+        private readonly IOptionsMonitor<FunctionsHostingConfigOptions> _functionsHostingConfigOptionsMonitor;
+        private readonly Func<Task> _stopApplication;
 
         private IOptions<WorkerConcurrencyOptions> _workerConcurrencyOptions;
         private IFunctionInvocationDispatcher _functionInvocationDispatcher;
         private System.Timers.Timer _timer;
-        private System.Timers.Timer _activationTimer;
         private ValueStopwatch _addWorkerStopwatch = ValueStopwatch.StartNew();
         private ValueStopwatch _logStateStopWatch = ValueStopwatch.StartNew();
-        private TimeSpan _activationTimerInterval = TimeSpan.FromMinutes(5);
         private bool _disposed = false;
+        private IDisposable _hostinConfigOnChange;
 
         public WorkerConcurrencyManager(
             IFunctionInvocationDispatcherFactory functionInvocationDispatcherFactory,
             IEnvironment environment,
             IOptions<WorkerConcurrencyOptions> workerConcurrencyOptions,
-            IFunctionsHostingConfiguration functionsHostingConfigurations,
+            IOptions<FunctionsHostingConfigOptions> functionsHostingConfigOptions,
```

Comment:
Do you need both `IOptions` and `IOptionsMonitor`? You can call `IOptionsMonitor.CurrentValue` to get the equivalent of `IOptions.Value`.

Reply:
Good point!